nservicebus/messaging/messages-events-commands.md

```diff
@@ -52,6 +52,16 @@ Note: Prior to version 7.2, messages had to be defined as a `class`. Defing them
 
 Generic message definitions (e.g. `MyMessage<T>`) are not supported. It is recommended to use dedicated, simple types for each message or to use inheritance to reuse shared message characteristics.
 
+### Solution structure
+
+Messages define the data contract between two endpoints.
+
+It's recommended to use a dedicated assembly for message definitions. By keeping messages in a separate assembly, the amount of information and dependencies shared between services is minimized. It is recomended to have a separate message assembly for every service. Due to that, service can [evolve its contracts](/nservicebus/messaging/evolving-contracts.md) without impacting other services in the system. Every message should be declared in the message assembly of the service owning that message. 
+
+Consideration should be given to thinking about how the contract assembly will be used by services. When certain events are subscribed by multiple endpoints managed by other teams, it might make sense to publish those contracts as a separate NuGet package. Depending on the use and frequency of changes it could make sense to seprate contracts into multiple assemblies. 
```

Comment:
```suggestion
Consideration should be given to how the contracts assembly will be used by services. When certain events are subscribed to by multiple endpoints managed by other teams, it might make sense to extract those contracts into a separate NuGet package. Depending on the use and the frequency of changes it could make sense to separate contracts into multiple assemblies. 
```